connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -435,6 +453,13 @@ public void testPollRedelivery() throws Exception {
         assertTaskMetricValue("batch-size-max", 1.0);
         assertTaskMetricValue("batch-size-avg", 0.5);
 
+        assertEquals(workerCurrentOffsets, Whitebox.<Map<TopicPartition, OffsetAndMetadata>>getInternalState(workerTask, "currentOffsets"));
```

Comment:
Do we have to probe internal fields to verify this change? Couldn't we examine the offsets given to `SinkTask::preCommit` or `Consumer::commitAsync` instead?

Reply:
I've also added an assertion on the offsets given to precommit above. Happy to remove this line

Reply:
Yeah, I think the assertions you added above cover this case nicely 👍